### ISSUE TITLE
What are the requirements to add parallel support ?

### ISSUE THREAD

**Issue by MoOx** [conradz/sauce-tap-runner#2]
I am looking for speed and have the energy to make a PR to add parallel support. SauceLabs support that. [Zuul](https://github.com/defunctzombie/zuul) for example, too.

Are you open for this enhancement ?
If yes, what are the blockers if any ? Or the difficulties to handle ?

**Comment by conradz** [comment]
Yes, definitely would be :+1: for this to support running tests in parallel. Not sure what the blockers are although I know I ran into some developing this. Basically just try running tests in parallel (removing the safeguard checks in the code) and see what fails.

**Comment by MoOx** [comment]
It seems with the current codebase, simply change the moment the tunnel is created (or at least make a pause when a creation is already started) should be enough to prevent duplicate tunnel creation.
Since saucelabs allow parallels jobs, this should be enough to be able to handle that.

**Comment by ericelliott** [comment]
Any resolution reached?

**Comment by conradz** [comment]
No, but a PR would be welcomed
